src/argilla/client/feedback/training/base.py

```diff
@@ -288,11 +290,47 @@ def generate_model_card(self, output_dir: str) -> "ArgillaModelCard":
             template_path=ArgillaModelCard.default_template_path,
         )
 
-        model_card_path = Path(output_dir) / "README.md"
-        model_card.save(model_card_path)
-        self._logger.info(f"Model card generated at: {model_card_path}")
+        if output_dir:
+            model_card_path = Path(output_dir) / "README.md"
+            model_card.save(model_card_path)
+            self._logger.info(f"Model card generated at: {model_card_path}")
+
         return model_card
 
+    def push_to_huggingface(self, repo_id: str, generate_card: Optional[bool] = True, **kwargs) -> None:
+        """Push your model to [huggingface's model hub](https://huggingface.co/models).
+
+        Args:
+            repo_id:
+                The name of the repository you want to push your model and tokenizer to.
+                It should contain your organization name when pushing to a given organization.
+            generate_card:
+                Whether to generate (and push) a model card for your model. Defaults to True.
+        """
+        if not kwargs.get("token"):
+            # Try obtaining the token with huggingface_hub utils as a last resort, or let it fail.
+            from huggingface_hub import HfFolder
+
+            if token := HfFolder.get_token():
+                kwargs["token"] = token
+
+            if not kwargs.get("token", None):
+                # One last check for the tests. We use a different env var name
+                # that the one gathered with HfFolder.get_token
+                if token := os.environ.get("HF_HUB_ACCESS_TOKEN", None):
```

Comment:
```suggestion
                if token :=  kwargs.get("token", os.environ.get("HF_HUB_ACCESS_TOKEN", None)):
```
this might be easier?